TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TransformationLoader.java

```diff
@@ -79,22 +80,48 @@ private IJsonTransformer configureTransformerFromConfig(Map<String, Object> c) {
         }
         throw new IllegalArgumentException("Could not find a provider named: " + key);
     }
+
     public IJsonTransformer getTransformerFactoryLoader(String newHostName) {
-        return getTransformerFactoryLoader(newHostName, null);
+        return getTransformerFactoryLoader(newHostName, null, null);
     }
 
-    public IJsonTransformer getTransformerFactoryLoader(String newHostName, String fullConfig) {
+    public IJsonTransformer getTransformerFactoryLoader(String newHostName, String userAgent, String fullConfig) {
         try {
             var loadedTransformers = getTransformerFactoryFromServiceLoader(fullConfig);
             return new JsonCompositeTransformer(Stream.concat(
                     loadedTransformers,
-                    Optional.ofNullable(newHostName).stream().map(HostTransformer::new)
+                    Stream.concat(
+                            Optional.ofNullable(userAgent).stream().map(UserAgentTransformer::new),
+                            Optional.ofNullable(newHostName).stream().map(HostTransformer::new))
             ).toArray(IJsonTransformer[]::new));
         } catch (JsonProcessingException e) {
             throw new IllegalArgumentException("Could not parse the transformer configuration as a json list", e);
         }
     }
 
+    @AllArgsConstructor
+    private static class UserAgentTransformer implements IJsonTransformer {
+        public static final String USER_AGENT = "user-agent";
+        private final String userAgent;
+
+        @Override
+        public Map<String,Object> transformJson(Map<String,Object> incomingJson) {
+            var headers = (Map<String, Object>) incomingJson.get(JsonKeysForHttpMessage.HEADERS_KEY);
+            var oldVal = headers.get(USER_AGENT);
+            if (oldVal != null) {
+                if (oldVal instanceof List) {
+                    // see https://www.rfc-editor.org/rfc/rfc9110.html#name-field-lines-and-combined-fi
+                    oldVal = String.join(", ", (List<String>) oldVal);
```

Comment:
I'm trying to rationalize in my head how this is working. So headers.get(USER_AGENT) doesn't give us a string but a list object. Why are we joining element of the list and appending `,`? It seems like if anything this would be a semicolon. Can you shed any light here?

Reply:
See the RFC link...
```
When a field name is repeated within a section, its combined field value consists of the list of corresponding field line values within that section, concatenated in order, with each field line value separated by a comma.
```

That's the reason that we're concatenating them with a comma.  I suppose that we should really just grab the last one.  I'm not sure if keeping these broken into multiple lines will let the target do filtering (a lot of http libraries do NOT handle multi-line headers properly).  If a user doesn't like this behavior, they could always leave user-agent empty & write their own transformer.  This all seems reasonable enough at the moment w/out more feedback.

Reply:
I'm still a little skeptical but sounds sufficient for now